baseapp/abci_utils.go

```diff
@@ -178,75 +178,57 @@ func NewDefaultProposalHandler(mp mempool.Mempool, txVerifier ProposalTxVerifier
 // FIFO order.
 func (h DefaultProposalHandler) PrepareProposalHandler() sdk.PrepareProposalHandler {
 	return func(ctx sdk.Context, req *abci.RequestPrepareProposal) (*abci.ResponsePrepareProposal, error) {
+		var maxBlockGas uint64
+		if b := ctx.ConsensusParams().Block; b != nil {
+			maxBlockGas = uint64(b.MaxGas)
+		}
+
+		txSelector := NewTxSelector(uint64(req.MaxTxBytes), maxBlockGas)
```

Comment:
@alexanderbez can you put the txSelector on the ProposalHandler and make it an interface + possible to set. This would be useful for us.

Reply:
100%

Reply:
@itsdevbear lmk if the pushed changes seem good to you.

Reply:
Nitpick but i'd do 

```
type TxSelector interface {
    SelectTxs(maxTxBytes, maxBlockGas uint64)
}
```

over 
```
txSelector func(maxTxBytes, maxBlockGas uint64) TxSelector`
````

moreso personal preference though

Reply:
I do think its clearer on what the intention of the interface is to be though.

Reply:
For sure!

Reply:
I just changed the interface method to `SelectTxForProposal(maxTxBytes, maxBlockGas uint64, memTx sdk.Tx, txBz []byte) bool`. Hope that's Ok with you.